tests/test_remote_door.py

```diff
@@ -12,118 +12,199 @@
 # Copyright: Intra2net AG and aexpect contributors
 # Author: Plamen Dimitrov <plamen.dimitrov@intra2net.com>
 #
-# selftests pylint: disable=C0111,C0111
+# selftests pylint: disable=C0111,C0111,W0613,R0913
 
 import os
 import glob
 import re
+import shutil
 import unittest
 import unittest.mock as mock
 
 from aexpect import remote_door
+from aexpect.client import RemoteSession
 
 
-@remote_door.run_remotely
-def add_one(number):
-    """A small decorated test function."""
-    return number + 1
+def _local_login(client, host, port, username, password, prompt,
+                 linesep="\n", log_filename=None, log_function=None,
+                 timeout=10, internal_timeout=10, interface=None):
+    return RemoteSession("sh", prompt=prompt, client=client)
 
 
-@mock.patch('aexpect.remote_door.remote', mock.MagicMock())
+def _local_copy(address, client, username, password, port, local_path,
+                remote_path, limit="", log_filename=None, log_function=None,
+                verbose=False, timeout=600, interface=None, filesize=None,
+                directory=True):
+    shutil.copy(local_path, remote_path)
+
+
+@mock.patch('aexpect.remote_door.remote.copy_files_to', _local_copy)
+@mock.patch('aexpect.remote_door.remote.wait_for_login', _local_login)
 class RemoteDoorTest(unittest.TestCase):
     """Unit test class for the remote door."""
 
     def setUp(self):
-        self.session = mock.MagicMock(name='session')
-        self.session.client = "ssh"
+        self.session = RemoteSession("sh", prompt="#", client="ssh")
+        os.mkdir(remote_door.REMOTE_PYTHON_PATH)
```

Comment:
In case this directory already exists the setup fails (eg. on a terminated previous execution. How about `if os.path.isdir` or `os.path.exists`? (I don't want you to cover all cases, I'm just suggesting to be lenient towards the most common failures)

Reply:
Yes, I was wandering about the same but then realized that the directory should never be there if we want to execute unit tests and this is a good form of fast fail. Otherwise we are in the risk of writing into a preexisting directory and lose somebody's data (even not replacing the entire folder). But I also went through some inconvenience because of this so you have a point and I might make it more tolerant albeit also slightly more dangerous.